coreneuron/io/lfp.hpp

```diff
@@ -0,0 +1,226 @@
+    F one_m_mu(1.0 - mu);
+    auto log_integral = [&q2, &dxn](F a, F b) {
+        if (std::abs(q2) < std::numeric_limits<F>::epsilon()) {
+            if (a * b <= 0) {
+                throw std::invalid_argument("Log integral: invalid arguments " + std::to_string(b) +
+                                            " " + std::to_string(a) +
+                                            ". Likely electrode exactly on the segment and "
+                                            "no flooring is present.");
+            }
+            return std::abs(std::log(b / a)) / dxn;
+        } else {
+            return std::log((b + std::sqrt(b * b + q2)) / (a + std::sqrt(a * a + q2))) / dxn;
+        }
+    };
+    if (delta <= 0.0) {
+        return f * log_integral(-mu, one_m_mu);
+    } else {
+        F sqr_delta(std::sqrt(delta));
+        F d1(mu - sqr_delta);
+        F d2(mu + sqr_delta);
+        F parts = 0.0;
+        if (d1 > 0.0) {
+            F b(std::min(d1, 1.0) - mu);
+            parts += log_integral(-mu, b);
+        }
+        if (d2 < 1.0) {
+            F b(std::max(d2, 0.0) - mu);
+            parts += log_integral(b, one_m_mu);
+        };
+        // complement
+        F maxd1_0(std::max(d1, 0.0)), mind2_1(std::min(d2, 1.0));
+        if (maxd1_0 < mind2_1) {
+            parts += 1.0 / radius * (mind2_1 - maxd1_0);
+        }
+        return f * parts;
+    };
+}
+
+enum LFPCalculatorType { LineSource, PointSource };
+
+/**
+ * \brief LFPCalculator allows calculation of LFP given membrane currents.
+ */
+template <LFPCalculatorType Ty, typename SegmentIdTy = int>
+struct LFPCalculator {
+    /**
+     * LFP Calculator constructor
+     * \tparam Point3Ds A vector of 3D points type
+     * \tparam Vector A vector of floats type
+     * \param comm MPI communicator
+     * \param seg_start all segments start owned by the proc
+     * \param seg_end all segments end owned by the proc
+     * \param radius fence around the segment. Ensures electrode cannot be
+     * arbitrarily close to the segment
+     * \param electrodes positions of the electrodes
+     * \param extra_cellular_conductivity conductivity of the extra-cellular
+     * medium
+     */
+    template <typename Point3Ds, typename Vector>
+    LFPCalculator(const Point3Ds& seg_start,
+                  const Point3Ds& seg_end,
+                  const Vector& radius,
+                  const std::vector<SegmentIdTy>& segment_ids,
+                  const Point3Ds& electrodes,
+                  double extra_cellular_conductivity)
+        : segment_ids_(segment_ids) {
+        if (seg_start.size() != seg_end.size()) {
+            throw std::invalid_argument("Different number of segment starts and ends.");
+        }
+        if (seg_start.size() != radius.size()) {
+            throw std::invalid_argument("Different number of segments and radii.");
+        }
+        double f(1.0 / (extra_cellular_conductivity * 4.0 * boost::math::constants::pi<double>()));
```

Comment:
As @cattabiani mentioned below  to be exact and use the `pi` value everywhere, `pi` should take the value which is read from `NEURON` instead of the constant value provided either by `boost` or `math.h`. I can work on that.

Reply:
I created related PR #494

Reply:
@iomaganaris , following my latest commit, pi is apparently equal to 0. Probably a merge issue on my side. Can you take a look? I restored a hardcoded value.